.appveyor.yml

```diff
@@ -6,39 +6,50 @@ platform: x64
 clone_depth: 5
 environment:
   APPVEYOR_SAVE_CACHE_ON_ERROR: true
-  CLCACHE_SERVER: 1
-  PACKAGES: berkeleydb boost-filesystem boost-multi-index boost-signals2 boost-test boost-thread libevent openssl rapidcheck zeromq double-conversion
+  CLCACHE_SERVER: 1 
   PATH: 'C:\Python37-x64;C:\Python37-x64\Scripts;%PATH%'
   PYTHONUTF8: 1
   QT_DOWNLOAD_URL: 'https://github.com/sipsorcery/qt_win_binary/releases/download/v1.4/Qt5.9.8_x64_static_vs2019.zip'
   QT_DOWNLOAD_HASH: 'f285cbb02bec3b3f3cc2621e3fa7d5edf0d6a66fa30c57859e583acda954ea80'
   QT_LOCAL_PATH: 'C:\Qt5.9.8_x64_static_vs2019'
   VCPKG_INSTALL_PATH: 'C:\tools\vcpkg\installed'
 cache:
-- C:\tools\vcpkg\installed
+- C:\tools\vcpkg\installed -> build_msvc\vcpkg-packages.txt
 - C:\Users\appveyor\clcache -> .appveyor.yml, build_msvc\**, **\Makefile.am, **\*.vcxproj.in
 - C:\Qt5.9.8_x64_static_vs2019
 install:
 - cmd: pip install --quiet git+https://github.com/frerich/clcache.git@v4.2.0
 # Disable zmq test for now since python zmq library on Windows would cause Access violation sometimes.
 # - cmd: pip install zmq
 # Powershell block below is to install the c++ dependencies via vcpkg. The pseudo code is:
-# 1. If the vcpkg install directory exists assume dependencies are installed and do nothing. To
-#    force a fresh install of the packages delete the job's appveyor cache.
-# 2. Otherwise:
+# 1. Check whether all the required vcpkg packages are installed.
+# 2. If all packages are installed continue with the build (if a fresh install
+#    of any package is required delete the job's appveyor cache).
+# 3. Otherwise:
 #    a. Update the vcpkg source (including port files) and build the vcpkg binary,
-#    b. Install the required packages.
+#    b. Install the missing packages.
 - ps: |
+      $env:PACKAGES = Get-Content -Path build_msvc\vcpkg-packages.txt
       cd c:\tools\vcpkg
-      if(!(Test-Path -Path ($env:VCPKG_INSTALL_PATH))) {
+      $isPackageMissing = $false
+      Foreach($reqdPackage in $env:PACKAGES.split()) {
```

Comment:
Is this loop still required? vcpkg\installed will be cleared when a package is added or removed to the text file